src/components/List.jsx

```diff
@@ -0,0 +1,71 @@
+import React from 'react';
+import { Switch } from './Switch';
+
+class List extends React.Component {
+  render() {
+    const MemoComp = React.memo((props) => {
```

Comment:
Essentially `React.memo` returns a component (with extended functionality), you don't want to create components (memoed or plain) inside `render` since it's going to be recreated on every `List` update.
Also, if you use `memo` show a situation when this memoization gives benefits (similar to what I was showing you on our call).